client/client-lib/src/lib.rs

```diff
@@ -430,23 +431,51 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
 }
 
 impl Client<UserClientConfig> {
-    pub async fn fetch_gateway(&self) -> Result<LightningGateway> {
-        // fetch gateway from db
-        if let Some(gateway) = self
-            .context
+    pub async fn fetch_registered_gateways(&self) -> Result<Vec<LightningGateway>> {
+        let gateways = self.context.api.fetch_gateways().await?;
+        let current_active = self.fetch_active_gateway().await;
+
+        if current_active.is_some() {
+            // We have an active gateway in this client. We should update the list of registered gateways with that information.
+            Ok(gateways
+                .into_iter()
+                .map(|mut gateway| {
+                    if gateway.node_pub_key == current_active.clone().unwrap().node_pub_key {
```

Comment:
Maybe you can remove the `current_active` and just do something like 
`if let Some(current_active) = self.fetch_active_gateway().await.take() {}` instead of `if current_active.is_some() {`
so you could do  `if gateway.node_pub_key == current_active.node_pub_key` and avoid the clone

Reply:
I'll post a gist as comment